### PR TITLE
Add Lumo and Material themes

### LINKED ISSUE #1591: D&D fix styles
child of vaadin/vaadin-grid-flow#374

Add Lumo and Material styles

Screenshot 2019-04-15 15 12 39

- The number indicator is shown only when more than 1 rows are dragged
- The number indicator is positioned relative to the pointer

### PR DESCRIPTION
**Pull request by tomivirkki**
Fixes #1591

---
This change is [Reviewable](https://reviewable.io/reviews/vaadin/vaadin-grid/1596)

### CODE REVIEW

File: theme/lumo/vaadin-grid-styles.html [hunk 1]
```diff
@@ -88,57 +88,88 @@
...
         left: 0;
+        height: calc(var(--_lumo-grid-border-width) + 2px);
         pointer-events: none;
+        background: var(--lumo-primary-color-50pct);
       }
 
-      [part~="row"][row-drag="below"] [part~="cell"]::before {
-        box-shadow: 0 3px 0 0 var(--lumo-primary-color-50pct);
+      :host([theme~="no-row-borders"]) [dragover] [part~="cell"]::after {
+        height: 2px;
```
Comment (yuriy-fix) [inline theme/lumo/vaadin-grid-styles.html]: **Suggestion:** Should we use `calc` for some lumo variable over here, like `calc(var(--lumo-size-xs) / 12)` or `calc(var(--lumo-space-xs) / 2)` ?

File: theme/lumo/vaadin-grid-styles.html [hunk 2]
```diff
@@ -88,57 +88,88 @@
...
       }
 
-      [part~="row"][row-drag="above"] [part~="cell"]::before {
-        box-shadow: 0 -3px 0 0 var(--lumo-primary-color-50pct);
+      [part~="row"][dragover="below"] [part~="cell"]::after {
+        top: 100%;
+        bottom: auto;
+        margin-top: -1px;
```
Comment (yuriy-fix) [inline theme/lumo/vaadin-grid-styles.html]: **Suggestion:** Should we use `calc` for some lumo variable over here, like `calc(var(--lumo-space-xs) / 4)` ?

File: theme/lumo/vaadin-grid-styles.html [hunk 3]
```diff
@@ -88,57 +88,88 @@
...
-        content: attr(dragged-row-count);
+        content: attr(dragstart);
         align-items: center;
         justify-content: center;
         box-sizing: border-box;
-        padding: 0.4em calc(0.5em + var(--lumo-border-radius-s) / 4);
-        color: var(--lumo-primary-contrast-color);
-        background-color: var(--lumo-primary-color);
-        border-radius: var(--lumo-border-radius-s);
+        padding: 0.3em calc(0.3em + var(--lumo-border-radius-s) / 4);
```
Comment (yuriy-fix) [inline theme/lumo/vaadin-grid-styles.html]: **Suggestion:** `--lumo-space-xs` or `--lumo-space-xs * 1.2`.

File: theme/material/vaadin-grid-styles.html [hunk 4]
```diff
@@ -121,6 +121,88 @@
...
+      }
+
+      [part~="row"][dragover] [part~="cell"]::after {
+        content: "";
+        position: absolute;
+        top: 0;
+        right: 0;
+        bottom: 0;
+        left: 0;
+        height: 3px;
```
Comment (yuriy-fix) [inline theme/material/vaadin-grid-styles.html]: **Suggestion:** something like `--lumo-size-xs / 8` ?

File: theme/material/vaadin-grid-styles.html [hunk 5]
```diff
@@ -121,6 +121,88 @@
...
+        pointer-events: none;
+        background: var(--material-primary-color);
+      }
+
+      [part~="row"][dragover="below"] [part~="cell"]::after {
+        top: 100%;
+        bottom: auto;
+        margin-top: -1px;
```
Comment (yuriy-fix) [inline theme/material/vaadin-grid-styles.html]: **Suggestion:** `calc(var(--lumo-space-xs) / 4)`?

File: theme/material/vaadin-grid-styles.html [hunk 6]
```diff
@@ -121,6 +121,88 @@
...
+        display: block;
+        position: absolute;
+        left: var(--_grid-drag-start-x);
+        top: var(--_grid-drag-start-y);
+        z-index: 100;
+        content: attr(dragstart);
+        align-items: center;
+        justify-content: center;
+        box-sizing: border-box;
+        padding: 4px;
```
Comment (yuriy-fix) [inline theme/material/vaadin-grid-styles.html]: **Suggestion:** `--lumo-space-xs`?

File: theme/material/vaadin-grid-styles.html [hunk 7]
```diff
@@ -121,6 +121,88 @@
...
+        color: var(--material-primary-contrast-color);
+        background-color: var(--material-error-color);
+        min-width: 24px;
```
Comment (yuriy-fix) [inline theme/material/vaadin-grid-styles.html]: **Suggestion:** `--lumo-size-xs`

File: theme/material/vaadin-grid-styles.html [hunk 8]
```diff
@@ -121,6 +121,88 @@
...
+        border-radius: 2px;
```
Comment (yuriy-fix) [inline theme/material/vaadin-grid-styles.html]: **Suggestion:** `--lumo-border-radius-s / 2`

File: src/vaadin-grid-drag-and-drop-mixin.html [hunk 9]
```diff
@@ -72,37 +72,49 @@
...
+
+            e.dataTransfer.setDragImage(row, e.clientX - rowRect.left, e.clientY - rowRect.top);
           }
 
           const items = this._isSelected(row._item) ? this.selectedItems.slice(0) : [row._item];
           e.dataTransfer.setData('text', this.formatDragData(items));
 
-          row.setAttribute('dragged-row-count', items.length);
-          requestAnimationFrame(() => row.removeAttribute('dragged-row-count'));
+          row.setAttribute('dragstart', '');
```
Comment (yuriy-fix) [inline src/vaadin-grid-drag-and-drop-mixin.html]: Was the naming already agreed? Sounds a little bit confusing to me. Not a blocker.

File: theme/material/vaadin-grid-styles.html (hunk 8 above)
Comment (tomivirkki) [reply theme/material/vaadin-grid-styles.html]: Shouldn't use lumo variables in material theme

File: theme/material/vaadin-grid-styles.html (hunk 7 above)
Comment (tomivirkki) [reply theme/material/vaadin-grid-styles.html]: Shouldn't use lumo variables in material theme

File: theme/material/vaadin-grid-styles.html (hunk 6 above)
Comment (tomivirkki) [reply theme/material/vaadin-grid-styles.html]: Shouldn't use lumo variables in material theme

File: theme/material/vaadin-grid-styles.html (hunk 5 above)
Comment (tomivirkki) [reply theme/material/vaadin-grid-styles.html]: Shouldn't use lumo variables in material theme

File: theme/material/vaadin-grid-styles.html (hunk 4 above)
Comment (tomivirkki) [reply theme/material/vaadin-grid-styles.html]: Shouldn't use lumo variables in material theme

File: theme/lumo/vaadin-grid-styles.html (hunk 3 above)
Comment (tomivirkki) [reply theme/lumo/vaadin-grid-styles.html]: Might be better to use `--lumo-space-xs` indeed. This pattern is copied & adjusted from https://github.com/vaadin/vaadin-lumo-styles/blob/master/badge.html#L13

File: theme/lumo/vaadin-grid-styles.html (hunk 1 above)
Comment (tomivirkki) [reply theme/lumo/vaadin-grid-styles.html]: The fixes 2px sizing comes from the static focus indicator sizing https://github.com/vaadin/vaadin-grid/blob/master/theme/lumo/vaadin-grid-styles.html#L75

File: theme/lumo/vaadin-grid-styles.html (hunk 2 above)
Comment (tomivirkki) [reply theme/lumo/vaadin-grid-styles.html]: The fixed sizing comes from the static focus indicator sizing https://github.com/vaadin/vaadin-grid/blob/master/theme/lumo/vaadin-grid-styles.html#L75

File: src/vaadin-grid-drag-and-drop-mixin.html (hunk 9 above)
Comment (tomivirkki) [reply src/vaadin-grid-drag-and-drop-mixin.html]: What sounds confusing about this name? Any other suggestions?

File: theme/lumo/vaadin-grid-styles.html (hunk 3 above)
Comment (tomivirkki) [reply theme/lumo/vaadin-grid-styles.html]: Done.

**Pull request merged**